### ISSUE TITLE
Example for fully dynamic configuration

### ISSUE THREAD

**Issue by Sitin** [envoyproxy/go-control-plane#591]
I am trying to dynamically mange Envoy.

The existing example sets the snapshot [only once](https://github.com/envoyproxy/go-control-plane/blob/main/internal/example/main/main.go#L60) when server starts.

It seems that only the first snapshot is applied. Any changes to it via `cache.SetSnapshot` do not take effect on Envoy.

It is possible that I am doing something wrong or do not fully understand how `go-control-plane` should work.

It would be nice to have an example that updates Envoy dynamically, not just using xDS to set Envoy configuration once and for all.

**Comment by Sitin** [comment]
Shame on me. Die to the stupid mistake I was working with the copy of the cache. Closing this.
